core/cache/namespace.go

```diff
@@ -54,17 +54,19 @@ func (n *namespaceStore) Start(ctx context.Context) {
 		defer ticker.Stop()
 
 		for {
-			list := &v1.NamespaceList{}
+			for name, c := range n.client.ClientsPool().Clients() {
+				list := &v1.NamespaceList{}
 
-			err := n.client.List(newCtx, list)
-			if err != nil {
-				n.logger.Error(err, "poll error")
-			}
+				err := c.List(newCtx, list)
+				if err != nil {
+					continue
```

Comment:
I'll add only logging, because even if one returned with error, it does not mean the rest will do the same. For example during fetching namespaces from a cluster that's unreachable for the moment, but in 1-2 mins it will be available again, the rest of the namespace lists from other clusters can be updated.